src/block_conversion_plugin.cpp

```diff
@@ -170,6 +170,14 @@ class block_conversion_plugin_impl : std::enable_shared_from_this<block_conversi
 
                   SILK_WARN << "Can't link new block " << *new_block;
 
+                  // Double check if it's duplicated ones.
+                  // We do not need to import block again during reconnection if it's in this cache.
+                  auto dup_block = std::find_if(native_blocks.begin(), native_blocks.end(), [&new_block](const auto& nb){ return nb.id == new_block->id; });
+                  if( dup_block != native_blocks.end() ) {
+                     SILK_WARN << "Receiving duplicated blocks " << new_block->id <<  " It's normal if it's caused by reconnection to SHiP.";
+                     return;
+                  }
+
```

Comment:
This check for duplicate native blocks is not needed.
If a duplicate block is sent to the block_conversion_plugin it will handle that as a fork and recreating the necessary EVM blocks.

Reply:
This is to avoid the problem we discussed last time: the fork process will trigger rewinding, but it will only reset the canonical head. So duplicated blocks will trigger a bunch of warnings.
https://github.com/eosnetworkfoundation/silkworm/issues/39